src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Rename/RenamerProjectTreeActionHandler.cs

```diff
@@ -61,6 +63,68 @@ internal partial class RenamerProjectTreeActionHandler : ProjectTreeActionHandle
             _operationProgressService = operationProgressService;
         }
 
+        public override async Task CopyAsync(IProjectTreeActionHandlerContext context, string filename, string folderToMove)
+        {
+            Requires.NotNull(context, nameof(Context));
+            Requires.NotNull(filename, nameof(filename));
+            Requires.NotNull(folderToMove, nameof(folderToMove));
+            // Get actions to update namespace
+            CodeAnalysis.Project? project = GetCurrentProject();
+
+            if (project is null)
+            {
+                return;
+            }
+
+            string folderPath = Path.GetDirectoryName(folderToMove);
+            folderPath = GetLastDirectory(folderPath);
+            string filenameWithExtension = Path.GetFileName(filename);
+            (bool result, Renamer.RenameDocumentActionSet? documentRenameResult) =
+                await GetRenameSymbolsActions(project, filename, filenameWithExtension, new List<string>(new string[]{ folderPath }));
+
+            await base.CopyAsync(context, filename, folderToMove);
+
+            if (!result || documentRenameResult == null)
+            {
+                return;
+            }
+
+            // Execute actions
+            _threadingService.RunAndForget(async () =>
+            {
+                Solution currentSolution = await PublishLatestSolutionAsync();
+
+                await _projectVsServices.ThreadingService.SwitchToUIThread();
+
+                WaitIndicatorResult<Solution> result = _waitService.Run(
+                    title: VSResources.Renaming_Type,
+                    message: "Update Namespace",
```

Comment:
this message will be changed

Reply:
The default provided from Roslyn is `Sync namespace to folder structure` if that's useful